### ISSUE TITLE
Finish up ROCAUC

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#252]
ROCAUC needs a few things to be finalized:

- [x] decision function bug (binary classifier) 
- [x] documentation 
- [x] add to gallery 
- [x] simplify example in docstring and switch to YB data instead of `sklearn.datasets`
- [x] add `multiclass` and `binary` fixtures from conftest.py (see [ClassificationReport](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_classification_report.py#L46)
- [x] remove [`X`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L44), [`yb`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L53), and [`ym`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L54) prefering to use above fixtures
- [x] remove [`load_binary_data`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L74) and [`load_multiclass_data`](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/tests/test_classifier/test_rocauc.py#L93) to use above fixtures 

Note the bug is as follows:

```
======================================================================
ERROR: Test ROCAUC with a binary classifier
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/benjamin/Repos/ddl/yellowbrick/tests/test_classifier/test_rocauc.py", line 110, in test_binary_rocauc
    s = visualizer.score(X_test, y_test)
  File "/Users/benjamin/Repos/ddl/yellowbrick/yellowbrick/classifier/rocauc.py", line 171, in score
    self.fpr[i], self.tpr[i], _ = roc_curve(y, y_pred[:,i], pos_label=c)
IndexError: too many indices for array
```

**Comment by derekflint** [comment]
+ 1

**Comment by rebeccabilbro** [comment]
@bbengfort - re: the decision function bug, I did a little research ([notebook here](https://github.com/rebeccabilbro/yellowbrick/blob/rocauc-refactor/examples/rebeccabilbro/rocauc_bug_research.ipynb)) into which classifiers and conditions actually seem to trigger the bug in the hopes that it might give us some ideas about how we should address it. 

In essence, it appears the bug is only triggered when the sklearn estimator has only a `decision_function` and not also a `predict_proba`. Estimators with both are fine, and estimators with only `predict_proba` also seem to be fine.

**Comment by bbengfort** [comment]
First I would sync your actual images with baseline images. From the yellowbrick project root, run:

```
$ python -m tests.images tests/test_classifier/test_rocauc.py 
```

Run the tests a couple of times to see if they pass; if they do, go ahead and push to a PR so we can see what happens on CI. 

If syncing the images does not fix the tests, then it's likely there is something random happening; look for anything in the tests that might have a `random_state` parameter (data generators, models, train test split, metrics, etc), and if it does set the `random_state` equal to some fixed number. Hopefully, that will solve the problem.

**Comment by rebeccabilbro** [comment]
Thanks @bbengfort - yeah, these errors were from after doing the image sync. I'll look into the `random_state` params!

**Comment by rebeccabilbro** [comment]
Looks like setting the `random_state` for `LinearSVC` and `RandomForestClassifier` did the trick!

**Comment by bbengfort** [comment]
Awesome!

**Comment by rebeccabilbro** [comment]
Closed with #540
